dimod/generators/wireless.py

```diff
@@ -0,0 +1,985 @@
+# Copyright 2023 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express 93or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+from collections import namedtuple
+from functools import wraps
+from itertools import product
+from typing import Literal, Optional, Tuple, Union
+
+import numpy as np
+
+import dimod
+
+__all__ = ['mimo', 'comp', ]
+
+mod_params = namedtuple("mod_params", ["bits_per_transmitter", 
+                                       "amplitudes", 
+                                       "transmitters_per_spin", 
+                                       "number_of_amps", 
+                                       "spins_per_symbol"])
+mod_config = {
+    "BPSK": mod_params(1, 1, 1, 1, 1),
```

Comment:
If we want to be even more explicit we could do
```suggestion
    "BPSK": mod_params(bits_per_transmitter=1, amplitudes=1, transmitters_per_spin=1, number_of_amps=1, spins_per_symbol=1),
```
but that might be getting long

Reply:
It would make these next five lines much longer and having the named parameters just above the value-setting lines should be clear enough